tests/time_evol.jl

```diff
@@ -1,65 +1,63 @@
+# Here we are trying to simlate the vaccuum oscillation term in the Rogerro Hamiltonian 
 let
-  N = 100 # number of sites
-  cutoff = 1E-8 # specifies a truncation threshold for the SVD in MPS representation
-  tau = 0.1 # time step 
+  N = 4 # number of sites
+  cutoff = 1E-14 # specifies a truncation threshold for the SVD in MPS representation
+  tau = 0.05 # time step 
   ttotal = 5.0 # total time of evolution 
+  del_x = 1E-3 # length of the box of interacting neutrinos at a site/shape function width of neutrinos in cm 
+  del_m2= 1.93E-23 #ergs squared
+  tolerance  = 5E-1 # acceptable level of error or deviation from an exact value or solution
 
   # Make an array of 'site' indices and label as s 
   # conserve_qns=true conserves the total spin quantum number "S"(in z direction) in the system as it evolves
-  s = siteinds("S=1/2", N; conserve_qns=false)  
-
-  # Make gates (1,2),(2,3),(3,4),... i.e. unitary gates which act on neighboring pairs of sites in the chain.
-  # Create an empty ITensors array that will be our Trotter gates
-  # gates acts as a single-particle operator that does not actually entangle pairs
-  gates = ITensor[] 
-
-  for j in 1:(N - 1)
-    #s1, s2 are neighbouring spin site/indices from the s array
-    s1 = s[j] 
-    s2 = s[j + 1]
-
-    # Hamiltonian operator oriented in the x direction of neigbouring spin sites. 
-    # "Sx" and "Id" are inbuilt ITensor operators in ITensors library and op function applies these ITensors to a set of neigboring site indices aka pairs s1, s2.
-    # hj operator acts on the pair system such that h at site index j and j+1 is computed separately and then summed on this 1D chain.
-    # Identity operator on adjacent index is added to form a pair that attaches to the gate operator, hence gates is effectively a one-particle operator.
-    hj = pi * (
-      op("Sx", s1)* op("Id", s2)  + op("Sx", s2) * op("Id", s1)
-    )
-
-    # make Trotter gate Gj that would correspond to each gate in the gate array of ITensors
-    Gj = exp(-im * tau / 2 * hj)  
-
-    # The push! function adds (appends) an element to the end of an array;
-    # ! performs an operation without creating a new object, (in a way overwites the previous array in consideration); 
-    # i.e. we append a new element Gj (which is an ITensor object representing a gate) to the end of the gates array.
-    push!(gates, Gj) 
-  end
-
-  # append! adds all the elements of a gates in reverse order (i.e. (N,N-1),(N-1,N-2),...) to the end of gates array.
-  append!(gates, reverse(gates)) 
+  s = siteinds("S=1/2", N; conserve_qns=true)  
+
+  # Initialize an array of ones for all N particles
+  mu = zeros(N)
+    
+  # Create an array of dimension N and fill it with the value 1/(sqrt(2) * G_F)
+  n = mu.* fill((del_x)^3/(sqrt(2) * G_F), N)
+    
+  # Create an array B with N elements. Each element of the array is a vector [0, 0, 1]
+  B = fill([0, 0, 1], N)
```

Comment:
The Hamiltonian in the old version points along the x direction. In order to keep the simulation unchanged, I think we need that B=[1,0,0], right?